Tests/CoreTests/EngineTests.swift

```diff
@@ -11,66 +11,78 @@ final class EngineTests: XCTestCase {
 
     let store = TestStore(
       initialState: Engine.State(
-        stats: [],
-        activity: .check(TestActivity.State.one)
+        stats: [:],
+        activity: .one,
+        core: CoreState()
       ),
       reducer: Engine(
-        activity: Activity(
-          checkActivity: TestActivity(),
-          otherActivity: EmptyReducer<Never, Never>()
-        ),
-        nextActivity: { .check(.two) }
+        activity: TestActivity(),
+        nextActivity: { _, _ in .two }
       )
     ) {
       $0.date = DateGenerator { now }
     }
 
-    await store.send(.check(.checkAnswer))
+    await store.send(.core(.checkAnswer))
 
     let result = ActivityResult<TestActivityType>(
       activityType: .one,
       date: now,
       status: .correct
     )
-    await store.receive(.check(.delegate(.storeResult(.correct)))) {
-      $0.stats.append(result)
+    await store.receive(.delegate(type: .one, action: .storeResult("1", .correct))) {
+      $0.stats["1"] = [result]
     }
 
-    await store.receive(.check(.delegate(.activityWasCompleted))) {
-      $0.activity = .check(.two)
+    await store.receive(.delegate(type: .one, action: .activityWasCompleted)) {
```

Comment:
In future we should move activity type from `.delegate` paramter to `Delegate.storeResult`. This test clearly shows that activity type is not needed to handle `. activityWasCompleted` delegate action